src/components/PipelineRun/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -313,6 +393,20 @@ const hasParentDep = (
       continue;
     }
     const depNode = nodes.find((n) => n.id === otherDep);
+    if (!depNode) {
+      // Try to find by original name (for matrix tasks)
+      const depNodeByOriginal = nodes.find((n) => {
+        const matrixTask = n.data.task as MatrixPipelineTaskWithStatus;
+        return matrixTask.originalName === otherDep;
+      });
+      if (
+        depNodeByOriginal?.runAfterTasks?.includes(dep) ||
+        hasParentDep(dep, depNodeByOriginal?.runAfterTasks || [], nodes)
+      ) {
+        return true;
+      }
+      continue;
```

Comment:
nit: I believe we don't need this `continue` here

Reply:
thanks for your review, I addressed both of your points